### DISCUSSION TITLE
Custom Time Scale for Arms Equivolume Chart

### DISCUSSION THREAD

**Discussion by luxsush** [d3/d3#3926] [Q&A] [answered]
Hi 

I am trying to create a custom scale for a Equivolume Chart where the candles width is varying and is proportional to the Volume traded on that candle

So the time scale is not equidistant but the tick widths vary per the volume in the candle as shown below

![image](https://github.com/user-attachments/assets/f5db8114-a665-4910-80b3-c445fe65cb4c)

I researched and found that I might have to use an Interpolator as shown below.  Was wondering if you could provide some high level guidance

`
var interpolatorFactory = function(a, b) {
    return function(t) {
        // Your custom scale function here, 
        // where you interpolate between a and b 
        // using the parameter t.
        return a * (1 - t) + b * t;
    };
};
var myScale = d3.scaleLinear().interpolate(interpolatorFactory);
// myScale is a standard d3 scale, so you can do 
// myScale.range([0, 5])
// myScale.domain([1, 3]) 
// etc
`

**Comment by luxsush** [comment, marked as answer]
I think I found this example to be the closest possible solution.  I can probably replace the color scale with a timescale here

https://observablehq.com/d/f7139ebca0286081

Thank you @Fil 

I have one query - how to do an invert of the custom weightedQuantileF function?

**Reply by Fil** [reply]
I've added the invert function to the notebook. Thanks for the suggestion
